ext/semian/tickets.c

```diff
@@ -1,12 +1,82 @@
 #include "tickets.h"
 
+// Update the ticket count for static ticket tracking
+static VALUE
+update_ticket_count(update_ticket_count_t *tc);
+
+static void
+initialize_tickets(int sem_id, int tickets, double quota);
+
+static int
+calculate_quota_tickets(int sem_id, double quota);
+
 static void
-initialize_tickets(int sem_id, int tickets);
+wait_for_init(int sem_id);
+
+void
+configure_tickets(int sem_id, int tickets, double quota, int should_initialize)
+{
+  int state;
+  update_ticket_count_t tc;
+
+  if (should_initialize) {
+    initialize_tickets(sem_id, tickets, quota);
+  }
+
+  if (quota > 0) {
+    tickets = calculate_quota_tickets(sem_id, quota);
+  } else {
+    wait_for_init(sem_id);
```

Comment:
the wait_for_init logic was extracted from the static ticket configuration.

It's only valid for static tickets, which is why it's separated out here. With quota it will cause an infinite wait (well, a timeout).